service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+
+import java.time.LocalDateTime;
+import java.time.format.FormatStyle;
+
+@Service
+public class CommonCaseDataExtractionService {
+    private final DateFormatterService dateFormatterService;
+
+    @Autowired
+    public CommonCaseDataExtractionService(DateFormatterService dateFormatterService) {
+        this.dateFormatterService = dateFormatterService;
+    }
+
+    public String getHearingTime(HearingBooking hearingBooking) {
+        String hearingTime;
+        final LocalDateTime startDate = hearingBooking.getStartDate();
+        final LocalDateTime endDate = hearingBooking.getEndDate();
+
+        if (hearingBooking.hasDatesOnSameDay()) {
+            // Example 3:30pm - 5:30pm
+            hearingTime = String.format("%s - %s",
+                formatTime(startDate), formatTime(endDate));
+        } else {
+            // Example 18 June, 3:40pm - 19 June, 2:30pm
+            hearingTime = String.format("%s - %s", formatDateTime(startDate),
+                formatDateTime(endDate));
+        }
+
+        return hearingTime;
+    }
+
+    public String getHearingDate(HearingBooking hearingBooking) {
+        String hearingDate = "";
+
+        // If they aren't on the same date return nothing
+        if (hearingBooking.hasDatesOnSameDay()) {
+            hearingDate = dateFormatterService.formatLocalDateToString(
+                hearingBooking.getStartDate().toLocalDate(), FormatStyle.LONG);
+        }
+
+        return hearingDate;
+    }
+
+    public String extractPrehearingAttendance(HearingBooking booking) {
+        LocalDateTime time = calculatePrehearingAttendance(booking.getStartDate());
+
+        return booking.hasDatesOnSameDay() ? formatTime(time) : formatDateTime(time);
+    }
+
+    private LocalDateTime calculatePrehearingAttendance(LocalDateTime dateTime) {
+        return dateTime.minusHours(1);
+    }
+
+    private String formatDateTime(LocalDateTime dateTime) {
+        return dateFormatterService.formatLocalDateTimeBaseUsingFormat(dateTime, "d MMMM, h:mma");
```

Comment:
As mentioned earlier I start to dislike the hardcoded patterns in random places :(